Added some missing engine command permissions.

Zumorica opened the pull request:
## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
Also CanViewVar is determined by whether you can execute the "vv" command or not.
Something to note is that content client-side commands cannot have permissions right now... This should probably be addressed somehow, but out of scope for now-- (see #5783)

Efruit commented:
This would totally break https://github.com/space-wizards/space-station-14/pull/5746 (and wouldn't work with [the related engine PR](https://github.com/space-wizards/RobustToolbox/pull/2323))

Zumorica commented:
summoning a superior power
@PJB3005

Efruit commented on Resources/engineCommandPerms.yml:
```diff
@@ -65,15 +123,20 @@
   - testlog
   - sudo
+  - scsi
+  - csi
```
I don't think this should be restricted, since it already requires a build with `CLIENT_SCRIPTING`

Efruit replied:
Because of the above, I think this should be `Debug` at max, or frankly, unrestricted. If anyone manages to get a build with it `CLIENT_SCRIPTING` on a public server, they've almost certainly undone any permissions checks.

Efruit commented on Resources/engineCommandPerms.yml:
```diff
@@ -23,13 +32,61 @@
+  - showchunkbb
+  - entfo
+  - sggcell
+  - ldrsc
+  - rldrsc
+  - rldshader
+  - rldloc
```
:thumbsdown:

Efruit replied:
Actually, aren't these restricted to debug builds too?

Efruit commented:
In fact, I think almost all the commands under `Debug` should be unrestricted. Not everyone troubleshooting client issues will be admins of the servers they're on, and we're hobbling the available tools significantly.

metalgearsloth commented:
If you leave Debug unrestricted you may as well leave them all unrestricted and let people wallhack with togglelight again.

Efruit commented on Resources/engineCommandPerms.yml:
```diff
@@ -23,13 +32,61 @@
+  - lsmonitor
+  - monitorinfo
+  - setmonitor
+  - watch
```
Shouldn't this be with `csi`?

PJB3005 replied:
Yes, it should be.

Efruit commented:
Again, the right solution for that is "don't send them things they can't see", but regardless.

So here's the list of `Debug`-gated commands in the PR (as of like 10 minutes ago).
```yaml
# Block
  - delete # yep
  - inrangeunoccluded # yep
  - lsgrid # yep
  - lsmap # yep
  - togglefov # yes
  - togglehardfov # yes
  - togglelight # yes
  - togglelightbuf # yes
  - toggleshadows # yes
  - loc # yep
  - showpos # yes
  - entfo # yes
  - sggcell # yes
  - dumpentities # yes
  - chunkinfo # yes
  - sendgarbage # yes
  - showanchored # yes
  - watch # again, w/ csi?

# Maybe
  - netaudit # maybe?
  - physics # maybe?
  - showislands # maybe?
  - lightbb # maybe?
  - vv # maybe?
  - showspritebb # maybe?
  - getcomponentregistration # maybe
  - showray # maybe
  - showchunkbb # maybe
  - scene # maybe
  - net_watchent # maybe
  - net_draw_interp # maybe
  - launchauth # maybe?
  - showvelocities # maybe
  - tilelookup # maybe
  - net_entityreport # maybe

# Don't know
  - addview # dunno
  - removeview # dunno
  - cldbglyr # doubt it
  - setinputcontext # probably not

# Do not block
  - showtime # no
  - testbed # no?
  - szr_stats # no
  - hwid # no
  - querymappaused # no
  - dmetamem # no
  - mem # no
  - listplayers # no? already shown in the lobby

# DO NOT BLOCK
  - monitor # NO
  - ldrsc # NO
  - rldrsc # NO
  - rldshader # NO
  - rldloc # NO
  - guidump # NO
  - uitest # NO
  - keyinfo # NO
  - lsmonitor # NO
  - monitorinfo # NO
  - setmonitor # NO
  - net_graph # NO
  - vram # NO
  - devwindow # NO

# why
  - setclipboard # why
  - getclipboard # why
  - fill # why
  - fuck # why
```

PaulRitter commented:
> Again, the right solution for that is "don't send them things they can't see", but regardless.

dang, why didn't i think of this. looking forward to your pr with an algo to performantly calculate if an entity is in someones fov and not in darkness for every player for every tick.

ofc the right solution is to not send them things they cant see, but you have to be somewhat pragmatic

metalgearsloth commented:
@Efruit leave reviews on the commands being addressed in the PR in the file rather than dumping a list of every command and your thoughts on it as this is just going to hold up legitimate permissions fixes.

Efruit commented:
@PaulRitter Be careful what you wish for. Are you SURE you want me touching PVS after how rich text turned out?

@metalgearsloth I explicitly did not do that because reviews don't work well for "move this line here"

> this is just going to hold up legitimate permissions fixes.

oh no, it'll take even longer for me to continue not having access to commands that I routinely used

what a pity

PJB3005 reviewed (commented):
Following commands really shouldn't be locked, IMO:
* `vram`
* `monitor`
* `setmonitor`
* `keyinfo`
* `getclipboard`
* `setclipboard`
* `szr_stats`
* `gcf`
* `lsasm`

These are all commands that have 0 chance of being abused in any way and could potentially be useful to help debug stuff (especially `monitor`). No reason to remove them.

PaulRitter commented:
> @PaulRitter Be careful what you wish for. Are you SURE you want me touching PVS after how rich text turned out?

no, i was trying to tell you that it isnt quite as trivial as "just dont send what they cant see lol"

> oh no, it'll take even longer for me to continue not having access to commands that I routinely used

well damn, suppose because you have to wait, the admins should wait too. solid logic

Efruit commented:
> no, i was trying to tell you that it isnt quite as trivial as "just dont send what they cant see lol"

too late, you've opened pandora's box
EVS coming soon™

> well damn, suppose because you have to wait, the admins should wait too. solid logic

I think you misread that message.

Zumorica commented:
> These are all commands that have 0 chance of being abused in any way and could potentially be useful to help debug stuff (especially `monitor`). No reason to remove them.

I've added all those except for `szr_stats` and `lsasm`. Those two are on the client but also on the server, and because our permission system is currently fucky that would mean giving anyone access to those commands on the server, if they modified their client somehow.

Also @PJB3005 what's your take on the CanViewVar changes? I changed this because I was told the permissions would be inconsistent otherwise, but I'd like your input on it.

PJB3005 commented:
> what's your take on the CanViewVar changes?

CanViewVar being behind vv? Fund it.

Efruit commented on Resources/engineCommandPerms.yml:
```diff
@@ -23,13 +39,55 @@
+  - fill
+  - dumpentities
+  - getcomponentregistration
+  - fuck
```
This needs to be publicly accessible, period.

Efruit commented on Resources/engineCommandPerms.yml:
```diff
@@ -23,13 +39,55 @@
+  - net_graph
+  - net_watchent
+  - net_draw_interp
+  - devwindow
```
Unrestricted, please.
